src/App.js

```diff
@@ -13,111 +14,100 @@ import NotificationsPortal from '@redhat-cloud-services/frontend-components-noti
     });
 
-  componentWillUnmount() {
-    if (typeof this.unregister === 'function') {
-      this.unregister();
-    }
-  }
-  async componentDidMount() {
-    insights.chrome.init();
-    insights.chrome?.hideGlobalFilter?.();
-    insights.chrome.identifyApp('remediations');
-    // wait for auth first, otherwise the call to RBAC may 401
-    await window.insights.chrome.auth.getUser().then((user) =>
-      this.setState({
-        hasSmartManagement: user.entitlements.smart_management.is_entitled,
-      })
-    );
-    getIsReceptorConfigured().then((isConfigured) =>
-      this.setState({
-        isReceptorConfigured: isConfigured.data.length > 0,
-      })
-    );
-    this.unregister = insights.chrome.on('APP_NAVIGATION', (event) => {
-      if (typeof event?.domEvent?.href === 'string') {
-        this.props.history.push('/');
-      }
-    });
-    window.insights.chrome
-      .getUserPermissions('remediations')
-      .then((remediationsPermissions) => {
-        const permissionList = remediationsPermissions.map(
-          (permissions) => permissions.permission
+  useEffect(() => {
+    let unregister;
+    if (chrome) {
+      chrome.identifyApp('remediations');
+      chrome?.hideGlobalFilter();
+
+      // wait for auth first, otherwise the call to RBAC may 401
+      chrome?.auth
+        .getUser()
+        .then((user) =>
+          setHasSmartManagement(user.entitlements.smart_management.is_entitled)
         );
-        if (
-          permissionList.includes(
-            'remediations:*:*' || 'remediations:remediation:*'
-          )
-        ) {
-          this.handlePermissionUpdate(true, true, true);
-        } else {
-          this.handlePermissionUpdate(
-            permissionList.includes(
-              'remediations:remediation:read' || 'remediations:*:read'
-            ),
-            permissionList.includes(
-              'remediations:remediation:write' || 'remediations:*:write'
-            ),
+
+      getIsReceptorConfigured().then((isConfigured) =>
+        setIsReceptorConfigured(isConfigured.data.length > 0)
+      );
+
+      unregister = chrome.on('APP_NAVIGATION', (event) =>
+        history.push(`/${event.navId}`)
+      );
+
+      chrome
+        .getUserPermissions('remediations')
+        .then((remediationsPermissions) => {
+          const permissionList = remediationsPermissions.map(
+            (permissions) => permissions.permission
+          );
+          if (
             permissionList.includes(
-              'remediations:remediation:execute' || 'remediations:*:execute'
+              'remediations:*:*' || 'remediations:remediation:*'
             )
-          );
-        }
-      });
-  }
-
-  render() {
-    const {
-      readPermission,
-      writePermission,
-      executePermission,
-      arePermissionLoaded,
-      isReceptorConfigured,
-      hasSmartManagement,
-    } = this.state;
+          ) {
+            handlePermissionUpdate(true, true, true, true);
```

Comment:
it seems the 4th argument is not needed (nor present in arguments disclosure for handlePermissionUpdate)
```suggestion
            handlePermissionUpdate(true, true, true);
```